crates/nu-protocol/src/engine/stack.rs

```diff
@@ -726,9 +741,29 @@ impl Stack {
             let path = nu_path::strip_trailing_slash(path);
             let value = Value::string(path.to_string_lossy(), Span::unknown());
             self.add_env_var("PWD".into(), value);
+            // Sync with PWD-per-drive
+            #[cfg(windows)]
+            {
+                let _ = self.pwd_per_drive.set_pwd(&path);
+            }
             Ok(())
         }
     }
+
+    // Helper stub/proxy for nu_path::expand_path_with::<P, Q>(path, relative_to, expand_tilde)
+    // Facilitates file system commands to easily gain the ability to expand PWD-per-drive
+    pub fn expand_path_with<P, Q>(&self, path: P, relative_to: Q, expand_tilde: bool) -> PathBuf
```

Comment:
I feel like it's a little weird that this API does *not* expand relative to what you passed in if it is a drive-relative path

Does Stack always have enough information to do PWD anyway? I forget if `EngineState::cwd` actually needs anything from there if stack is provided. I would rather have an API that doesn't make you provide the CWD if it's going to be derived from the stack in some cases anyway.

Reply:
Originally, when the file system command code calls nu_path::expand_path_with<P, Q>(path, relative_to, expand_tilde), it provides relative_to as a hint and uses expand_tilde as an indicator. The Stack acts as a stub or proxy here, primarily to check if, on Windows, the path can be expanded by PWD-per-drive. If the path is relative to a drive, PWD-per-drive ensures it is expanded to an absolute path and returns it. If the path is not relative to drive, it forwards the call to nu_path::expand_path_with<P, Q>(path, relative_to, expand_tilde) to complete the task as before.